apisix/plugins/traffic-split.lua

```diff
@@ -258,18 +258,21 @@ end
 local function new_rr_obj(weighted_upstreams)
     local server_list = {}
     for i, upstream_obj in ipairs(weighted_upstreams) do
-        if not upstream_obj.upstream then
+        if not upstream_obj.upstream and not upstream_obj.upstream_id then
             -- If the `upstream` object has only the `weight` value, it means
             -- that the `upstream` weight value on the default `route` has been reached.
-            -- Need to set an identifier to mark the empty upstream.
-            upstream_obj.upstream = "empty_upstream"
+            -- Mark empty upstream services in the plugin, but with weight.
+            upstream_obj.upstream = "plugin#upstream#is#empty"
+            server_list[upstream_obj.upstream] = upstream_obj.weight
         end
 
-        if type(upstream_obj.upstream) == "table" then
+        if upstream_obj.upstream_id then
```

Comment:
Where is the definition of the `upstream_id`?

Reply:
It has been defined here: https://github.com/apache/apisix/pull/3512/files#diff-95a2018f002f15f5fb0714075b5ef27a1db113c186bfe20ae705f6023080b38fR100